examples/linear_model/plot_poisson_regression_non_normal_loss.py

```diff
@@ -145,7 +145,11 @@
         ("preprocessor", linear_model_preprocessor),
         ("regressor", DummyRegressor(strategy="mean")),
     ]
-).fit(df_train, df_train["Frequency"], regressor__sample_weight=df_train["Exposure"])
+).fit(
+    df_train.drop(columns=["Frequency"]),
```

Comment:
Just for confirmation, you want to make `linear_model_preprocessor` preprocess `df_train` & `df_test` only once for all 3 linear models (Dummy, Ridge, Poisson) and drop `Pipeline`, right? I will try the changes locally.

Reply:
I would use
```python
y_train = df_train.pop("Frequency")
y_test = df_test.pop("Frequency")
```
because I find it clean and it avoids data leakage from the beginning.

Reply:
@lorentzenchr Thanks for suggestion. I reverted to the original setup which lets the preprocessor to deal with columns, but am definitely open to implement improvements if needed.